Quickfixes for release

hal-ler opened the pull request:
To solve #522.

DustinCampbell commented on build.cake:
```diff
@@ -428,8 +428,8 @@ Task("OnlyPublish")
-            // No need to package for <!win7> + net451
-            else if (!runtimeShort.Contains("win7-") && framework.Equals("net451"))
+            // No need to package for <!win> + net451
+            else if (!runtimeShort.Contains("win-") && framework.Equals("net451"))
```
Ha! Oops. :smile:

DustinCampbell commented:
:+1:

david-driscoll commented:
The travis change is simple, just fix https://github.com/OmniSharp/omnisharp-roslyn/blob/dev/.travis.yml#L40

hal-ler commented:
Changed ifdef DNX451 in Nuget and Roslyn.Csharp folders to ifdef NET451.
Removed ifdef DNX451 in MSBuild folder (since it only ever builds for NET451).

hal-ler commented:
Mono archive renaming now only applies on Travis. This allows users to create archives for Mono on *NIX without being restricted to Ubuntu.

hal-ler commented:
Travis now pushes all archives (cross-compilation already controlled by PopulateRuntime and only triggered when desired).
